actions_helper/commands/deregister_task_definition.py

```diff
@@ -0,0 +1,61 @@
+from typing import Optional
+
+import click
+from botocore.client import BaseClient
+
+from actions_helper.outputs import CreateTaskDefinitionOutput
+from actions_helper.utils import set_error
+
+
+def deregister_task_definition(
+    ecs_client: BaseClient,
+    cluster: str,
+    service: str,
+    production_task_definition_output: Optional[CreateTaskDefinitionOutput] = None,
+    local_task_definition_output: Optional[CreateTaskDefinitionOutput] = None,
+    preflight_task_definition_output: Optional[CreateTaskDefinitionOutput] = None,
+):
+    (primary_deployment_definition_arn,) = (
+        deployment["taskDefinition"]
+        for deployment in ecs_client.describe_services(
+            cluster=cluster,
+            services=[service],
+        )["services"][0]["deployments"]
+        if deployment["status"] == "PRIMARY"
+    )
+
+    click.echo(f"{primary_deployment_definition_arn=}")
+
+    production_task_definition_to_deregister = local_task_definition_to_deregister = (
+        preflight_task_definition_to_deregister
+    ) = None
+
+    fail_pipeline = True
+    if (
+        production_task_definition_output
+        and primary_deployment_definition_arn == production_task_definition_output.latest_task_definition_arn
+    ):
+        # Do not deregister task definition for initial deployment
+        if production_task_definition_output.previous_task_definition_arn:
+            production_task_definition_to_deregister = production_task_definition_output.previous_task_definition_arn
+            local_task_definition_to_deregister = local_task_definition_output.previous_task_definition_arn
+            preflight_task_definition_to_deregister = preflight_task_definition_output.previous_task_definition_arn
+        fail_pipeline = False
+    else:
+        production_task_definition_to_deregister = production_task_definition_output.latest_task_definition_arn
+        local_task_definition_to_deregister = local_task_definition_output.latest_task_definition_arn
+        preflight_task_definition_to_deregister = preflight_task_definition_output.latest_task_definition_arn
```

Comment:
What happens if  `production_task_definition_output = None`? The way I see it is that in the else statement, `production_task_definition_output.latest_task_definition_arn` will be called, which should raise an exception. 

Can you please check this?